yt_dlp/extractor/instagram.py

```diff
@@ -28,7 +28,8 @@
 
 def _pk_to_id(media_id):
     """Source: https://stackoverflow.com/questions/24437823/getting-instagram-post-url-from-media-id"""
-    return encode_base_n(int(media_id.split('_')[0]), table=_ENCODING_CHARS)
+    pk = int(media_id.split('_')[0]) if type(media_id) is str else media_id
+    return encode_base_n(pk, table=_ENCODING_CHARS)
```

Comment:
Not a big fan of doing the type checking/conversion inside the method. Can you give an example where we pass `media_id` as an `int`?

Reply:
Will elaborate below!

Reply:
Let's just do this then
```py
    pk = int(str(media_id).split('_')[0])
    return encode_base_n(pk, table=_ENCODING_CHARS)

```